src/components/themr.js

```diff
@@ -80,30 +85,49 @@ export default (componentName, localTheme, options = {}) => (ThemedComponent) =>
         : {}
     }
 
-    getTheme() {
-      return this.props.composeTheme === COMPOSE_SOFTLY
-        ? { ...this.getContextTheme(), ...config.localTheme, ...this.getNamespacedTheme() }
-        : themeable(themeable(this.getContextTheme(), config.localTheme), this.getNamespacedTheme())
+    getTheme(props) {
+      return props.composeTheme === COMPOSE_SOFTLY
+        ? {
+          ...this.getContextTheme(),
+          ...config.localTheme,
+          ...this.getNamespacedTheme(props)
+        }
+        : themeable(
+          themeable(this.getContextTheme(), config.localTheme),
+          this.getNamespacedTheme(props)
+        )
+    }
+
+    calcTheme(props) {
+      const { composeTheme } = props
+      return composeTheme
+        ? this.getTheme(props)
+        : this.getThemeNotComposed(props)
+    }
+
+    componentWillReceiveProps(nextProps) {
```

Comment:
How do you handle context here? Seems like component won't react on changing its theme in context. Even without pure-rendering optimization.